runtime/dynlink.c

```diff
@@ -229,6 +220,85 @@ void caml_free_shared_libs(void)
     caml_dlclose(shared_libs.contents[--shared_libs.size]);
 }
 
+static value list_of_ext_table(struct ext_table* tbl)
+{
+  CAMLparam0();
+  CAMLlocal3(list, tmp, str);
+  int i;
+  for (i = tbl->size - 1; i >= 0; i--) {
+    str = caml_copy_string(tbl->contents[i]);
```

Comment:
Alas, `caml_shared_libs_path` contains `char_os` pointers, so (slightly tediously!) this needs to be `caml_copy_string_of_os` for `caml_shared_libs_path` and `caml_copy_string` for `caml_prim_name_table`.

(and apologies, because I missed this in my first review)